skopt/optimizer/optimizer.py

```diff
@@ -154,6 +155,7 @@ def __init__(self, dimensions, base_estimator="gp",
         n_jobs = acq_optimizer_kwargs.get("n_jobs", 1)
         self.acq_optimizer_kwargs = acq_optimizer_kwargs
 
+        dimensions = normalize_dimensions(dimensions, estimator=base_estimator)
```

Comment:
If we go ahead with my idea from https://github.com/scikit-optimize/scikit-optimize/pull/420#issuecomment-313610646 this is the line to delete.